src/app/devices/device-toolbar/device-toolbar.component.ts

```diff
@@ -85,4 +85,12 @@ export class DeviceToolbarComponent implements OnInit {
     this.devicesService.startwebSocket.next(true)
     await this.router.navigate([`/devices/${this.deviceId}/${path}`])
   }
+
+  async navigateToKvm (path: string): Promise<void> {
```

Comment:
why does this need to be a separate function?

Reply:
While navigating between kvm and sol pages, when using same navigateTo function due to subscriptions to eventEmitters websocket connections were established before calling ngDestroy.

Reply:
Removed navigateToKvm function and added the logic in navigateTo function